File: php/Abstraction_Q1/Employee.php
<?php
abstract class Employee
{
    private $name;
    private $employeeId;

    public function __construct($name, $employeeId)
    {
        $this->name = $name;
        $this->employeeId = $employeeId;
    }

    protected function getName()
    {
        return $this->name;
    }

    protected function getEmployeeId()
    {
        return $this->employeeId;
    }

    public abstract function calculatePay();
}
?>

File: pyhton/Abstraction_Q1/Employee.py
from abc import ABC, abstractmethod

class Employee(ABC):
    def __init__(self, name, employee_id):
        self._name = name
        self._employee_id = employee_id

    def get_name(self):
        return self._name

    def get_employee_id(self):
        return self._employee_id

    @abstractmethod
    def calculate_pay(self):
        pass